### PR TITLE
maxheap

### PR DESCRIPTION
**Pull request by ryantheboi**
Please quickly review the implementation of the Max Heap data structure.
Also, please run Planters.java through the tests again.

### CODE REVIEW

File: hw/hw1/Prob3-Writeup.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+* Line 2 contains integers s1, s2, ..., sp, separated by spaces.
+* Line 3 contains integers t1, t2, ..., tr, separated by spaces.
+* The integers are positive and fit in int.
+* p and r are non-negative and not larger than 2,500,000.
+* n = p + r
+
+#### Code:
+```
+# Glossary
+MH = Min Heap
```
Comment (zedoax) [inline hw/hw1/Prob3-Writeup.md]: Delete?  You don't use it anywhere else in the writeup

File: hw/hw1/Prob3-Writeup.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+# Setup - read input and store them into appropriate data structures
+1. read in the first line of input into int p, int r
+2. read in the second line of input into int[] s
+3. read in the third line of input into int[] t
+4. initialize and build Max Heap heapS for array s
+5. initialize and build Max Heap heapT for array t
+
+# Main Loop -  Compare the max roots between the two heaps in order to prioritize the placement of the largest plant
+               into the largest available pot
+6.   count = 0
```
Comment (zedoax) [inline hw/hw1/Prob3-Writeup.md]: \noindent

File: hw/hw1/prob3/Planters.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import java.util.Scanner;
 
 /**
- * Created by Ryan Chung on 9/11/2018.
+ * Created by Ryan Chung on 9/14/2018.
  */
```
Comment (zedoax) [inline hw/hw1/prob3/Planters.java]: Turn this into a docstring
/**
 * Planters - description
 * @author - Ryan Chung
 * @author - Elijah ...
 * @author - Ryan ...
 **/

Comment (zedoax) [reply hw/hw1/prob3/Planters.java]: Also you can add @ modified - 09/14/18 if you want the date

File: hw/hw1/Prob3-Writeup.md (hunk 1 above)
Comment (ryantheboi) [reply hw/hw1/Prob3-Writeup.md]: o sht

File: hw/hw1/Prob3-Writeup.md (hunk 2 above)
Comment (ryantheboi) [reply hw/hw1/Prob3-Writeup.md]: oops

File: hw/hw1/prob3/Planters.java (hunk 3 above)
Comment (ryantheboi) [reply hw/hw1/prob3/Planters.java]: done

**Comment by zedoax** [comment]
After looking over the file, all the extra documentation needed is the purpose for the heaps at the top

File: hw/hw1/prob3/Planters.java [hunk 4]
```diff
@@ -1,8 +1,15 @@
 import java.util.Scanner;
 
 /**
- * Created by Ryan Chung on 9/14/2018.
- */
+ @modified - 09/14/18
+ Planters - Uses 2 max heaps to compare the max roots between the two heaps created from the list of plants that need
```
[the remark this answers is not in the document]
Comment (ryantheboi) [reply hw/hw1/prob3/Planters.java]: oops done

File: hw/hw1/prob3/Planters.java [hunk 5]
```diff
@@ -28,16 +35,16 @@ public static void main( String[] arg ) {
...
         while ( !heapS.isEmpty() && !heapT.isEmpty() ){
             maxPlant = heapS.extractMax();
             maxPot = heapT.extractMax();
-            if ( maxPlant < maxPot ){
+            if ( maxPlant < maxPot ){ // will the largest plant fit in the largest pot?
                 maxPot.add( maxPlant );
                 count++;
             }
-            else{
+            else{ // the largest plant cannot find a complementary larger pot, therefore output NO
```
[the remark this answers is not in the document]
Comment (ryantheboi) [reply hw/hw1/prob3/Planters.java]: done

File: hw/hw1/prob3/Planters.java [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
- into the largest available pot
-
- @author - Ryan Chung
- @author - Elijah Bendinsky
- @author - Joshua Ehling
+ * @modified - 09/14/18
+ * Planters - Uses 2 max heaps to compare the max roots between the two heaps created from the list of plants that need
+ * new pots and from the list of available pots in order to prioritize the placement of the largest plant
+ * into the largest available pot
+ *
```
Comment (zedoax) [inline hw/hw1/prob3/Planters.java]: don't leave whitespace in the docstring
Also modified should be the last element in the docstring

File: hw/hw1/prob3/Planters.java [hunk 7]
```diff
@@ -35,16 +35,22 @@ public static void main( String[] arg ) {
         heapT.buildHeap( t );
 
         int count = 0;
-        int maxPlant; // the root of the max heap for plants that need new pots
-        int maxPot; // the root of the max heap for available pots
+        int maxPlant; // the root of the max heap for plants that need new pots and available pots, respectively
```
Comment (zedoax) [inline hw/hw1/prob3/Planters.java]: move this to the line above.  DO NOT COMMENT on the same line as your code except in special cases

**Pull request merged**